Release 10/13/2017

ctro opened the pull request.

mgwalker commented on src/components/Form/Accordion/Accordion.test.jsx:
```diff
@@ -333,4 +346,40 @@ describe('The accordion component', () => {
+        func: () => {},
+        expected: false
+      }
+    ]
```
See the comment here about additional tests:  https://github.com/18F/e-QIP-prototype/pull/92#discussion_r137691465

mgwalker commented on src/components/Form/Telephone/Telephone.jsx:
```diff
@@ -645,10 +645,32 @@ Telephone.errors = [
+              !!props.international.second &&
+              !!props.international.third
+            break
+          default:
```
See comment here: https://github.com/18F/e-QIP-prototype/pull/92#discussion_r137692792

bmallred commented on api/model/form/branch.go:
```diff
@@ -32,3 +36,59 @@ func (entity *Branch) Valid() (bool, error) {
+	var err error
+	if err = context.CreateTable(&Branch{}, options); err != nil {
+		return entity.ID, err
+	}
```
Yeah I think so. It would also help to make it a little bit more abstract in the case of switching out ORMs

bmallred commented on api/model/form/benefit.go:
```diff
@@ -1,29 +1,142 @@
 }
 
 // Unmarshal bytes in to the entity properties.
 func (entity *Benefit) Unmarshal(raw []byte) error {
```
As the data comes via the HTTP request the body is in binary format already. The `Unmarshal` allows us to transform the JSON in to our intended structure(s).

We currently don't have a use for a `Marshal` function yet (I think!) but there may be a need when sending saved data back to the client. Currently the JSON encoding functions we have are handling this.

Did that answer your question?

bmallred commented on api/handlers/root_test.go:
```diff
@@ -20,12 +20,6 @@ func TestRoot(t *testing.T) {
-	//if rec.Body.String() != expected {
-	//	t.Errorf("handler returned unexpected body: got %v want %v", rec.Body.String(), expected)
-	//}
-
```
Ha! We just updated that as well and I glossed over that test case... I'll fix that!

bmallred commented:
In reference to https://github.com/18F/e-QIP-prototype/pull/92#discussion_r137776673...

I believe we have one more PR to review for the accordion refactoring and then will be able to remove the unnecessary code which was in place so we didn't break functionality in areas not updated.

ctro commented:
One thought I want to leave here:
I like this part of the workflow we inherited where we set up PRs from the fork for review as weekly "Releases".  It keeps review more focused and keeps mental overhead lower for reviewers.

However I don't like that it's more difficult to address feedback with specific commits.  We don't want the TrueTandem devs to add overhead to their process by addressing feedback in 18F's repo, then pulling that back into the fork -- this project sees so many changes that... that seems... yuck.

Currently the TT devs are addressing the feedback here ^^ in their fork.  In the next couple days I'll probably merge this PR and set up a new release PR -- the next PR will address the feedback in this PR.  This also doesn't seem great.

@mgwalker do you have any ideas for improving this process?  I don't have a ton of experience working with forks.

mgwalker commented:
Yeah, that's definitely more complicated than ideal.  I don't know of anything better offhand.  Would it be useful if I opened `chore` issues on the TT repo for things that don't need to be fixed _right now_?  That is, if something needs to be fixed before the PR can be merged, all of the feedback can live in the PR.  But everything else can begin life as a chore issue in the TT fork, and the PR can just reference that?  And a chore could be addressed alongside a more substantive issue rather than being standalone, which might make things a little simpler?

I really don't know, just spitballing. @bmallred please feel free to throw your ideas in here too!  I don't want to send you and your team on a lot of back-and-forth between repos.  :)
